docs/structured-streaming-programming-guide.md

```diff
@@ -2251,13 +2251,13 @@ When the streaming query is started, Spark calls the function or the object’s
 
 - The close() method (if it exists) is called if an open() method exists and returns successfully (irrespective of the return value), except if the JVM or Python process crashes in the middle.
 
-- **Note:** The partitionId and epochId in the open() method can be used to deduplicate generated data 
-  when failures cause reprocessing of some input data. This depends on the execution mode of the query. 
-  If the streaming query is being executed in the micro-batch mode, then every partition represented 
-  by a unique tuple (partition_id, epoch_id) is guaranteed to have the same data. 
-  Hence, (partition_id, epoch_id) can be used to deduplicate and/or transactionally commit 
-  data and achieve exactly-once guarantees. However, if the streaming query is being executed 
-  in the continuous mode, then this guarantee does not hold and therefore should not be used for deduplication.
+- **Note:** Spark doesn't guarantee same output for (partitionId, epochId) on failure, so deduplication
+  cannot be achieved with (partitionId, epochId). e.g. source provides different number of
+  partitions for some reason, Spark optimization changes number of partitions, etc.
+  Refer SPARK-28650 for more details. `epochId` can still be used for deduplication, but there's less
+  benefit to leverage this, as the chance for Spark to successfully write all partitions and fail to checkpoint
```

Comment:
Using epoch seems not quite useful given the description. Should we maybe just remove it out?

Reply:
I tend to agree that, but if we drop the case, the result of `open` is just going to be always `false`. As we are waiting for @zsxwing to review this, let's hear the voice on this. If @zsxwing doesn't provide any voice I'll just remove it out from both here and javadoc.

Reply:
I got the response below (https://github.com/apache/spark/pull/25407#discussion_r313034261) and looks OK to remove. I'll remove it altogether.